### PR TITLE
Added StateStore

### PR DESCRIPTION
**Pull request by tenbradley**

### CODE REVIEW

**Comment by ciss1995** [comment]
LGTM, I'll just add tests.

File: state/stateStore.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class StateStore {
```
Comment (m6jones) [inline state/stateStore.js:1]: needs to be exported

File: state/stateStore.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	constructor() {
+		this._states = new Map();
+	}
+
+	add(state) {
```
Comment (m6jones) [inline state/stateStore.js:6]: needs to be async and add at the start `await state.tokenRefresh()`

File: state/stateStore.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		if (!state || !state.entityId || !state.token.toString) {
+			return;
+		}
+
+		this._initContainer(state.entityId, state.token);
+	}
+
+	get(entityID, token) {
```
Comment (m6jones) [inline state/stateStore.js:14]: needs to be async and add at the start `await state.tokenRefresh()`

File: state/stateStore.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		const lowerCaseEntityId = entityID.toLowerCase();
+		return this.has(entityID, token) && this._states.get(token.toString()).get(lowerCaseEntityId).keys().next().value;
+	}
+
+	has(entityID, token) {
```
Comment (m6jones) [inline state/stateStore.js:19]: needs to be async and add at the start `await state.tokenRefresh()`

File: state/stateStore.js (hunk 3 above)
Comment (tenbradley) [reply state/stateStore.js:14]: How can I await the tokenRefresh at the start when I don't have the state yet?

**Review by ciss1995 — APPROVED** [review]

**Pull request merged**